Source/deviceinfo/DeviceInfo.cpp

```diff
@@ -611,106 +614,574 @@ class DeviceInfoLink : public WPEFramework::RPC::SmartInterfaceType<WPEFramework
+        if(_platformName.IsSet())
+        {
+            strncpy(buffer, _platformName.Value().c_str(), _platformName.Value().length());
+            *length = _platformName.Value().length();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_platform_name(buffer,length);
+            if(result == Core::ERROR_NONE)
+            {
+                _platformName = buffer;
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_maximum_output_resolution(deviceinfo_output_resolution_t* value)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_maxOutputResolution.IsSet())
+        {
+            *value = _maxOutputResolution;
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_maximum_output_resolution(value);
+            if(result == Core::ERROR_NONE)
+            {
+                _maxOutputResolution = *value;
+            }
+        }
+        return result;
+    }
+
+    uint32_t Deviceinfo_output_resolutions(deviceinfo_output_resolution_t value[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_outputResolution.size() > 0)
+        {
+            std::copy(_outputResolution.begin(), _outputResolution.end(), value);
+            *length = _outputResolution.size();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_output_resolutions(value, length);
+            if(result == Core::ERROR_NONE)
+            {
+                _outputResolution.insert(_outputResolution.begin(), value, value + (*length));
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_video_outputs(deviceinfo_video_output_t value[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_videoOutput.size() > 0)
+        {
+            std::copy(_videoOutput.begin(), _videoOutput.end(), value);
+            *length = _videoOutput.size();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_video_outputs(value, length);
+            if(result == Core::ERROR_NONE)
+            {
+                _videoOutput.insert(_videoOutput.begin(), value, value + (*length));
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_audio_outputs(deviceinfo_audio_output_t value[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_audioOutput.size() > 0)
+        {
+            std::copy(_audioOutput.begin(), _audioOutput.end(), value);
+            *length = _audioOutput.size();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_audio_outputs(value, length);
+            if(result == Core::ERROR_NONE)
+            {
+                _audioOutput.insert( _audioOutput.begin(), value, value + (*length));
+            }
+        }
+        return result;
+    }
+
+private:
+
+    void setHdrSupport(const bool& hdr)
```

Comment:
Why "const bool&" and not just "const bool"? (yes perhaps the bool in the second version is not what you are used to but we do that even for copy by value params to indicate that they are not supposed to change although that is would not happen anyway (and it saved us from some bugs when somebody by accident tried to change them and was not aware that would not propagate to the caller)